src/parquet/schema/descriptor.h

```diff
@@ -54,24 +53,32 @@ class ColumnDescriptor {
     return primitive_node_->physical_type();
   }
 
+  LogicalType::type logical_type() const {
+    return primitive_node_->logical_type();
+  }
+
   const std::string& name() const {
     return primitive_node_->name();
   }
 
   bool is_required() const {
-    return max_definition_level_ == 0;
+    return primitive_node_->is_required();
   }
 
   bool is_optional() const {
-    return max_definition_level_ > 0;
+    return primitive_node_->is_optional();
   }
 
   bool is_repeated() const {
-    return max_repetition_level_ > 0;
+    return primitive_node_->is_repeated();
   }
```

Comment:
Since this didn't cause test failures, it may be worth trying to add a failing test case